.githooks/pre-commit

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -euxo pipefail
+
+KT_FILES="$(git --no-pager diff --name-status --no-color --cached | awk '$1 != "D" && $2 ~ /\.kts|\.kt/ { print $2}')"
+
+if [ -z "$KT_FILES" ];
+then
+  echo "No Kotlin staged files."
+else
+  echo "Running ktlint over these files:"
+  echo "$KT_FILES"
+
+  ./gradlew ktlintFormat -PinternalKtlintGitFilter="$KT_FILES"
+
+  echo "Completed ktlint run."
+
+  echo "Starting detekt run"
+  ./gradlew detekt
+  echo "Completed detekt run"
+
+  echo "$KT_FILES" | while read -r file; do
+      if [ -f $file ]; then
```

Comment:
Why is this extra check neccessary?

Reply:
The if checks if a file does actually exist before adding it via git.

I'm not entirely sure which in which edge-case this would become relevant. I was of course "heavily inspired" from something I found on the internet when writing this hook. If for some reason ktlintFormat has entirely deleted a file this if statement would become relevant.

I would prefer to leave it in just in case it does actually does cover some edge case.

Reply:
😄 Ok, it doesn't hurt in any case.